secrets/secrets.go

```diff
@@ -0,0 +1,69 @@
+package secrets
+
+import (
+	"errors"
+	"sync"
+)
+
+var (
+	// ErrNotImplemented default secrets in OSD
+	ErrNotImplemented = errors.New("Not Implemented")
+)
+
+type Secrets interface {
+	// Login create session with secret store
+	Login(secretType string, secretConfig map[string]string) error
+	// DefaultSecretKey  sets the cluster wide secret key
+	SetDefaultSecretKey(secretKey string, override bool) error
+	// GetDefaultSecretKey returns cluster wide secret key's value
+	GetDefaultSecretKey() (interface{}, error)
+	// CheckSecretLogin validates session with secret store
+	CheckLogin() error
+	// SetSecret sets secret key against data
+	Set(key string, value interface{}) error
+	// GetSecret retrieves the data for key
+	Get(key string) (interface{}, error)
+}
+
+type Manager struct {
+	Secrets
+	lock sync.Mutex
+}
+
+func NewSecretManager(sec Secrets) *Manager {
+	return &Manager{
+		Secrets: sec,
+	}
+}
+
+type nullSecretManager struct {
```

Comment:
nit: can we call this nullSecrets instead of manager, as it is an implementation of Secrets interface?